src/wms/src/main/java/org/geoserver/wms/map/RenderedImageMapOutputFormat.java

```diff
@@ -505,10 +508,39 @@ public RenderedImageMap produceMap(final WMSMapContent mapContent, final boolean
         
         onBeforeRender(renderer);
 
-        // setup the timeout enforcer (the enforcer is neutral when the timeout is 0)
+        //Timeout on the smallest nonzero value of the WMS timeout and the timeout format option
+        //If both are zero then there is no timeout
         int maxRenderingTime = wms.getMaxRenderingTime() * 1000;
+        int localMaxRenderingTime = 0;
```

Comment:
Can we encapsulate all these calculation in their own method? It seems they would just return the max rendering time anyways.